File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/model/EscapeProfile.java
package uk.gov.justice.digital.hmpps.riskprofiler.model;

import io.swagger.annotations.ApiModelProperty;
import lombok.*;

import javax.validation.constraints.NotBlank;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString
public class EscapeProfile extends RiskProfile {

    @ApiModelProperty(value = "Indicates offender is on the escape list", example = "true", position = 1)
    private boolean activeEscapeList;

    @ApiModelProperty(value = "Indicates offender is an escape risk", example = "true", position = 2)
    private boolean activeEscapeRisk;

    @ApiModelProperty(value = "Active escape risk alerts", position = 3)
    private List<Alert> escapeRiskAlerts;

    @ApiModelProperty(value = "Active escape list alerts", position = 4)
    private List<Alert> escapeListAlerts;

    public RiskType getRiskType() {
        return RiskType.ESCAPE;
    }

    @Builder(builderMethodName = "escapeBuilder")
    public EscapeProfile(@NotBlank String nomsId, @NotBlank String provisionalCategorisation, boolean activeEscapeList, boolean activeEscapeRisk, List<Alert> escapeListAlerts, List<Alert> escapeRiskAlerts) {
        super(nomsId, provisionalCategorisation);
        this.activeEscapeList = activeEscapeList;
        this.activeEscapeRisk = activeEscapeRisk;
        this.escapeListAlerts = escapeListAlerts;
        this.escapeRiskAlerts = escapeRiskAlerts;
    }
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/model/EscapeProfile.kt
package uk.gov.justice.digital.hmpps.riskprofiler.model

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank

data class EscapeProfile(
  val nomsId: @NotBlank String,
  var provisionalCategorisation: @NotBlank String,
  @field:Schema(
    title = "Indicates offender is on the escape list",
    example = "true",
  ) val activeEscapeList: Boolean,

  @field:Schema(
    title = "Indicates offender is an escape risk",
    example = "true",
  ) val activeEscapeRisk: Boolean,

  @field:Schema(title = "Active escape risk alerts") val escapeRiskAlerts: List<Alert>?,

  @field:Schema(title = "Active escape list alerts") val escapeListAlerts: List<Alert>?

) : RiskProfile(nomsId, provisionalCategorisation) {

  constructor() : this("", "", false, false, null, null)

  override fun getRiskType(): RiskType {
    return RiskType.ESCAPE
  }
}
